lib/jxl/modular/encoding/encoding.cc

```diff
@@ -196,6 +196,61 @@ Status DecodeModularChannelMAANS(BitReader *br, ANSSymbolReader *reader,
           }
         }
       }
+    } else if (predictor == Predictor::Gradient && offset == 0 &&
+               multiplier == 1 && reader->RleOnly()) {
+      JXL_DEBUG_V(8, "Gradient RLE (fjxl) very fast track.");
+      const intptr_t onerow = channel.plane.PixelsPerRow();
+      uint32_t run = 0;
+      uint32_t v = 0;
+      pixel_type sv = 0;
+      for (size_t y = 0; y < channel.h; y++) {
+        pixel_type *JXL_RESTRICT r = channel.Row(y);
+
+#if true
+        const pixel_type *JXL_RESTRICT rtop = (y ? channel.Row(y - 1) : r - 1);
+        const pixel_type *JXL_RESTRICT rtopleft =
+            (y ? channel.Row(y - 1) - 1 : r - 1);
+        {
+          size_t x = 0;
```

Comment:
Maybe clearer to remove the conditionals here

Reply:
Yes, simplified it - for x=0, clampedgradient boils down to just top :)